setup.py

```diff
@@ -17,11 +17,16 @@
 import sys
 
 # Always prefer setuptools over distutils
-from setuptools import setup, find_packages
+from setuptools import setup, find_packages, __version__ as setuptoolsversion
 # To use a consistent encoding
 from codecs import open
 from os import path
 
+#require at least setuptools 20.2 for PEP 508 conditional dependency support
+MIN_SETUPTOOLS_VERSION = (20, 2)
+if tuple(int(x) for x in setuptoolsversion.split('.')[:2]) < MIN_SETUPTOOLS_VERSION:
+    sys.exit("setuptools %s.%s or later is required." % MIN_SETUPTOOLS_VERSION)
```

Comment:
some people will see this and create an issue in our github issue tracker asking how to resolve this issue

I suggest you add ` To fix it, try running this: pip install "setuptools>=20.2"`.